### ISSUE TITLE
Regression : Can't use unicode charaters with "shell" module in 1.8.4

### ISSUE THREAD

**Issue by clonedagain** [ansible/ansible#10536]
This task used to work in 1.8.2 but doesn't work anymore in 1.8.4:

```
    - name: Les depôts privés doivent être déployés
      sudo_user: "{{ odoo_user }}"
      sudo: true
      shell: |
        if [ -d /home/{{ odoo_user }}/{{ item }}/ ]
        then
          echo __OK__ # le dépôt git existe déjà, on ne l écrase pas
        else
          git clone git@git:/srv/git/odoo/{{ item }}.git /home/{{ odoo_user }}/{{ item }} -b 8.0 -o team
        fi
      with_items: private_repos
      register: git
```

Trace when running the playbook with -vvvv:

```
failed: [devls-erp] => (item=private-fdl) => {"failed": true, "item": "private-fdl", "parsed": false}
SUDO-SUCCESS-pumhevaydwuwyciswsbyutijngpighzz
OpenSSH_6.6.1, OpenSSL 1.0.1f 6 Jan 2014
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 19: Applying options for *
debug1: auto-mux: Trying existing master
debug2: fd 3 setting O_NONBLOCK
debug2: mux_client_hello_exchange: master version 4
debug3: mux_client_forwards: request forwardings: 0 local, 0 remote
debug3: mux_client_request_session: entering
debug3: mux_client_request_alive: entering
debug3: mux_client_request_alive: done pid = 26256
debug3: mux_client_request_session: session request sent
debug1: mux_client_request_session: master session id: 2
Traceback (most recent call last):
  File "<stdin>", line 2064, in <module>
  File "<stdin>", line 195, in main
  File "<stdin>", line 1677, in run_command
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 59: ordinal not in range(128)
debug3: mux_client_read_packet: read header failed: Broken pipe
debug2: Received exit status from master 1
```

Then it works if I remove all non-ascii characters from the script.

**Comment by abadger** [comment]
Found the issue -- we're getting a byte string instead of a unicode string there.  Not sure where it's coming from but in v1, we can just fix it at the point we need to use it.  So I've got a fix that does just that.

Trying to write an integration test to go along with it so that things don't regress under v2.

**Comment by abadger** [comment]
Note -- this won't be in 1.9.0 but I'm going to add it to the stable-1.9 branch so it can be in a potential 1.9.1 release.

# Closing This Ticket

Hi!

We believe recent commits (likely detailed above) should resolve this question or problem for you.

This will also be included in the next major release.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular
issue is resolved.

Thank you!

**Comment by abadger** [comment]
And here's the same fix for v2: 60f972d

**Comment by abadger** [comment]
Oh -- and as for why it's a byte str... I realized that we're in module_utils/basic.py here.  Modules deal in byte str.  Only the controller side deals with unicode strings internally.
